ShopAppTests/App/Main/Views/SearchTitleViewSpec.swift

```diff
@@ -0,0 +1,302 @@
+            textField = self.findView(withAccessibilityLabel: "textField", in: view) as! UITextField
+            underLineView = self.findView(withAccessibilityLabel: "underline", in: view)
+            backButton = self.findView(withAccessibilityLabel: "back", in: view) as! UIButton
+            cartButtonView = self.findView(withAccessibilityLabel: "cart", in: view)
+            clearButton = self.findView(withAccessibilityLabel: "clear", in: view) as! UIButton
+            underlineLeftMargin = underLineView.superview!.constraints.filter({ $0.accessibilityLabel == "underlineLeft" }).first
+            underlineRightMargin = underLineView.superview!.constraints.filter({ $0.accessibilityLabel == "underlineRight" }).first
+        }
+        
+        describe("when view initialized") {
+            it("should have correct intrinsic content size") {
+                expect(view.intrinsicContentSize) == UILayoutFittingExpandedSize
+            }
+            
+            it("should have correct title of clear button") {
+                expect(clearButton.title(for: .normal)) == "Button.Clear".localizable
+            }
+            
+            it("should have correct attributed placeholder of text field") {
+                expect(textField.attributedPlaceholder?.string.hasSuffix("Placeholder.Search".localizable)) == true
+            }
+            
+            it("should have correct text alignment of text field") {
+                expect(textField.textAlignment) == NSTextAlignment.center
+            }
+            
+            it("should have correct background color of underline view") {
+                expect(underLineView.backgroundColor) == Colors.underlineDefault
+            }
+            
+            it("should have correct constants of underline view's constraints") {
+                expect(underlineLeftMargin.constant) == 55
+                expect(underlineRightMargin.constant) == 55
+            }
+            
+            it("should have hidden back button") {
+                expect(backButton.isHidden) == true
+            }
+            
+            it("should have unhidden cart button view") {
+                expect(cartButtonView.isHidden) == false
+            }
+            
+            it("should have hidden clear button") {
+                expect(clearButton.isHidden) == true
+            }
+            
+            it("should have cart button view without button") {
+                expect(cartButtonView.subviews.isEmpty) == true
+            }
+        }
+        
+        describe("when cart bar button item updated") {
+            beforeEach {
+                view.updateCartBarItem()
+            }
+            
+            it("needs to add button to cart button view") {
+                expect(cartButtonView.subviews.count) == 1
+            }
+            
+            it("needs to remove old button and add new one to cart button view on next time") {
+                view.updateCartBarItem()
```

Comment:
You call this method in `beforeEach`

Reply:
We need to call this method again because in this case we'll cover all code in this method.